src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -1007,7 +1006,13 @@ void UserInfoPopup::fetchSevenTVAvatar(const HelixUser &user)
                             this->getFilename(hash.result().toHex());
 
                         this->saveCacheAvatar(data, filename);
-                        this->setSevenTVAvatar(filename);
+
+                        bool hasFocus = (QApplication::focusWidget() == nullptr);
+                        if (hasFocus)
+                        {
+                            this->avatarUrl_ = URI;
+                            this->setSevenTVAvatar(filename);
+                        }
```

Comment:
`QApplication::focusWidget()` may not have the intended behaviour on all platforms (at least not for me). Instead, a `nullptr` check on the avatar button itself would be better as the button will be a  `nullptr` when closed.
```suggestion
                        if (this->ui_.avatarButton != nullptr)
                        {
                            this->avatarUrl_ = URI;
                            this->setSevenTVAvatar(filename);
                        }
```

Reply:
I agree.

Reply:
I tested my branch on Windows and Arch (https://streamable.com/e/f5y3ke) and it seemed to work fine. However your fix is much better. Thank you very much! :)

Reply:
I'm assuming it was a WM-related issue